### PR TITLE
[wxwidgets] Fix find dependency pcre2

### LINKED ISSUE #24851: wxWidgets build failure - dlls only copied in release profile CMAKE builds, debug profile builds unable to run (Duplicate of 21827)
See: https://github.com/microsoft/vcpkg/issues/21827 for sections on Host Environment, To Reproduce,. 

Issue 2187 was supposed to be closed in the following pull request: https://github.com/microsoft/vcpkg/pull/24047, but using the latest available wxWidgets port, the problem is still seen.

Like before, the test project builds/runs successfully in the release project, but it does not run successfully in the x64 debug project. It appears to be copying more of the .dll files into the build output, but not "pcre2-16.dll". So when run, the .exe reports "The code execution cannot proceed because pcre2-16.dll was not found. Reinstalling the program may fix this problem."

**Failure Logs**
```
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\vcpkg-cmake-x64-dbg-release-issue.exe'. Symbols loaded.
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\ntdll.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\kernel32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\KernelBase.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\vcruntime140d.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\vcruntime140_1d.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\wxbase316ud_vc_custom.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\wxmsw316ud_core_vc_custom.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\ucrtbased.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\ucrtbased.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Unloaded 'C:\Windows\System32\ucrtbased.dll'
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\user32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\user32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Unloaded 'C:\Windows\System32\user32.dll'
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\win32u.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\gdi32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\shell32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\gdi32full.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\msvcp_win.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\msvcp_win.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Unloaded 'C:\Windows\System32\msvcp_win.dll'
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\ucrtbase.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\ucrtbase.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Unloaded 'C:\Windows\System32\ucrtbase.dll'
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\shlwapi.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\comdlg32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\msvcrt.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\combase.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\ole32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\rpcrt4.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\advapi32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\SHCore.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\sechost.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\oleaut32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.22000.120_none_9d947278b86cc467\comctl32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\version.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\winspool.drv'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\winmm.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\zlibd1.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\msvcp140d.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\oleacc.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\uxtheme.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\jpeg62.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\libpng16d.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\tiffd.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Windows\System32\msimg32.dll'. 
'vcpkg-cmake-x64-dbg-release-issue.exe' (Win32): Loaded 'C:\Users\Steph\source\vcpkg-cmake-x64-dbg-release-issue\out\build\x64-Debug\liblzma.dll'. 
The thread 0x21b8 has exited with code 0 (0x0).
The thread 0x23d8 has exited with code 0 (0x0).
The thread 0x7a38 has exited with code 3221225781 (0xc0000135).
The program '[32588] vcpkg-cmake-x64-dbg-release-issue.exe' has exited with code 3221225781 (0xc0000135) 'A dependent dll was not found'.
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Currently, only the release version of the dependency pcre2 is found during the configuration of wxwidgets, fix this.

Fixes #24851.

### CODE REVIEW

File: ports/wxwidgets/fix-pcre2.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+-find_library(PCRE2_LIBRARIES
+-    NAMES pcre2-${PCRE2_CODE_UNIT_WIDTH}
+-    HINTS ${PC_PCRE2_LIBDIR}
+-          ${PC_PCRE2_LIBRARY_DIRS}
+-)
+-
+ include(FindPackageHandleStandardArgs)
++include(SelectLibraryConfigurations)
++find_library(PCRE2_LIBRARY_DEBUG NAMES pcre2-${PCRE2_CODE_UNIT_WIDTH}d PATH_SUFFIXES lib REQUIRED)
++find_library(PCRE2_LIBRARY_RELEASE NAMES pcre2-${PCRE2_CODE_UNIT_WIDTH} PATH_SUFFIXES debug/lib REQUIRED)
```
Comment (dg0yt) [inline ports/wxwidgets/fix-pcre2.patch:18]: The path suffixes are wrong w.r.t. to the variable names and lib names.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/wxwidgets/fix-pcre2.patch (hunk 1 above)
Comment (JackBoosY) [reply ports/wxwidgets/fix-pcre2.patch:18]: Maybe we can remove `PATH_SUFFIXES` and just use `NAMES` to find the correct library.

Comment (dg0yt) [reply ports/wxwidgets/fix-pcre2.patch:18]: You can omit `PATH_SUFFIXES` if release and debug use different `NAMES` for all triplets.

Comment (JackBoosY) [reply ports/wxwidgets/fix-pcre2.patch:18]: Finally I fixed this using pkgconfig, will push my commit later.

**Review by LilyWangLL — APPROVED** [review]

**Comment by dg0yt** [comment]
For x64-linux, the config tool (`tools/wxwidgets/wx-config`) doesn't report any pcre2 libs. x64-linux is static. I guess there will be a linking issue as soon as pcre2 functions from wxwidgets are actually used. But I can't answer now if this is a regression.

**Comment by JackBoosY** [comment]
The value `pcre` currently left in `wx-config` is from `libsecret-1.pc`->`glib-2.0.pc` in the system:
```
root@usr:/home/usr/work/24874/vcpkg# find /usr/lib/x86_64-linux-gnu/pkgconfig/ -name *.pc | xargs grep "pcre"
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcre.pc:Name: libpcre
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcre.pc:Libs: -L${libdir} -lpcre
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcreposix.pc:Name: libpcreposix
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcreposix.pc:Description: PCREPosix - Posix compatible interface to libpcre
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcreposix.pc:Libs: -L${libdir} -lpcreposix
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcreposix.pc:Requires.private: libpcre
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcre16.pc:Name: libpcre16
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcre16.pc:Libs: -L${libdir} -lpcre16
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcre32.pc:Name: libpcre32
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcre32.pc:Libs: -L${libdir} -lpcre32
/usr/lib/x86_64-linux-gnu/pkgconfig/libselinux.pc:Requires.private: libsepol libpcre
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcrecpp.pc:Name: libpcrecpp
/usr/lib/x86_64-linux-gnu/pkgconfig/libpcrecpp.pc:Libs: -L${libdir} -lpcre -lpcrecpp
/usr/lib/x86_64-linux-gnu/pkgconfig/glib-2.0.pc:Requires.private: libpcre
/usr/lib/x86_64-linux-gnu/pkgconfig/glib-2.0.pc:Libs.private: -pthread  -lpcre
```

**Comment by JackBoosY** [comment]
@dg0yt 
The remaining `pcre` is caused by wxwidgets using the default toolkit gtk3 in linux, it will look for _gtk+-3.0.pc_ in the system, and then use the other installed libraries in systems including `pcre`.
Then I tried adding `gtk` as a dependency of wxwidgets, but I get an error:
```
CMake Error at ports/wxwidgets/portfile.cmake:28 (message):
  Port freetype must not be installed when building wxwidgets:x64-linux.
```
Seems like this problem is hard to fix at the moment?
Can we ignore this issue for now?

**Comment by dg0yt** [comment]
If pcre2 is properly referenced in `wx-config` again, this PR is okay.

Everything else is unrelated:
- I never complained about `pcre`, the old version of the lib.
- You cannot use `gtk` (version 4!) for `wxwidgets`. That's why we need `gtk3`, #22201.

**Comment by JackBoosY** [comment]
The default toolkits which we can pick is:
```cmake
elseif(UNIX)
    set(wxDEFAULT_TOOLKIT gtk3)
    set(wxTOOLKIT_OPTIONS gtk2 gtk3 gtk4 motif qt)
    set(wxPLATFORM UNIX)
```
We can use qt but it will take more and more build time.

**Comment by dg0yt** [comment]
> The default toolkits which we can pick is:

I know. Qt might be a future feature... but it is pointless: Why use a cross-platform framework on top of another cross-platform framework? I think that developers expect the default, which is gtk3 for Linux.

**Comment by ras0219-msft** [comment]
> set(wxTOOLKIT_OPTIONS gtk2 gtk3 gtk4 motif qt)

I notice that gtk4 is on that list -- progress!

**Comment by ras0219-msft** [comment]
This LGTM though, thanks!

**Comment by dg0yt** [comment]
> > set(wxTOOLKIT_OPTIONS gtk2 gtk3 gtk4 motif qt)
> 
> I notice that gtk4 is on that list -- progress!

It is on the list, but it simply won't work soon - not even with wxWidgets 3.4.

**Review by Playgirlkaybraz11 — COMMENTED** [review]
#39474 #33955 #39548 #39588 ##39513

**Pull request merged**